graphrag/index/progress/load_progress_reporter.py

```diff
@@ -19,12 +19,13 @@ def load_progress_reporter(reporter_type: str = "none") -> ProgressReporter:
     -------
     ProgressReporter
     """
-    if reporter_type == "rich":
-        return RichProgressReporter("GraphRAG Indexer ")
-    if reporter_type == "print":
-        return PrintProgressReporter("GraphRAG Indexer ")
-    if reporter_type == "none":
-        return NullProgressReporter()
-
-    msg = f"Invalid progress reporter type: {reporter_type}"
-    raise ValueError(msg)
+    match reporter_type:
+        case "rich":
```

Comment:
You could use the ReporterType enum in here instead of strings

Reply:
Done